gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -53,13 +53,19 @@
   // Enable / disable state store
   public static final String STATE_STORE_ENABLED = "state.store.enabled";
   // DB state store configuration
+  public static final String STATE_STORE_DB_JDBC_DRIVER_KEY = "state.store.db.jdbc.driver";
+  public static final String DEFAULT_STATE_STORE_DB_JDBC_DRIVER = "com.mysql.jdbc.Driver";
+  // min idle time for eviction
+  public static final String STATE_STORE_DB_CONN_MIN_EVICTABLE_IDLE_TIME_KEY =
+      "state.store.db.conn.min.evictable.idle.time";
+  public static final long DEFAULT_STATE_STORE_DB_CONN_MIN_EVICTABLE_IDLE_TIME = 5 * 60 * 1000;
   public static final String STATE_STORE_DB_URL_KEY = "state.store.db.url";
   public static final String STATE_STORE_DB_USER_KEY = "state.store.db.user";
   public static final String STATE_STORE_DB_PASSWORD_KEY = "state.store.db.password";
   public static final String STATE_STORE_DB_TABLE_KEY = "state.store.db.table";
   public static final String DEFAULT_STATE_STORE_DB_TABLE = "gobblin_job_state";
-  public static final String STATE_STORE_DB_COMPRESS_KEY = "state.store.db.compress";
-  public static final boolean DEFAULT_STATE_STORE_DB_COMPRESS = true;
+  public static final String STATE_STORE_DB_COMPRESS_VALUE_KEY = "state.store.db.compress.value";
```

Comment:
As I like to say, dots.are.not.substitutes.for.spaces. We should use dots only for scoping. Unfortunately, we have a bunch of legacy settings that use that pattern but we should not keep using it.

I suggest using state.store.db.compressedValues or state.store.db.enableCompressedValues